src/main/editDelete/preProcess.ts

```diff
@@ -50,6 +31,74 @@ const injectAttributes: MapCallback<PartialWord, Word> = (word, i) => ({
   fileName: 'PLACEHOLDER FILENAME',
 });
 
+const constructWord = (
+  word: string,
+  startTime: number,
+  duration: number,
+  outputStartTime: number,
+  key: string
+) => {
+  const newWord: Word = {
+    word,
+    startTime: roundToMs(startTime),
+    duration: roundToMs(duration),
+    outputStartTime: roundToMs(outputStartTime),
+    deleted: false,
+    key,
+    fileName: 'PLACEHOLDER FILENAME',
+  };
+
+  return newWord;
+};
+
+/**
+ * Adds spaces between words which represent the silence between each word
+ * @param result The output being constructed
+ * @param word The current element of words
+ * @param index The index of word in words
+ * @param words The list of words being reduced
+ * @returns The updated transcript with a silence after word
+ */
+let TOTAL_DURATION = 0;
+const addSpaces = (
+  result: Word[],
+  word: Word,
+  index: number,
+  words: Word[]
+) => {
+  const spaceChar = ' ';
+
+  // is the first word
+  if (index === 0) {
+    result.push(
```

Comment:
The idea with reduce is that it's immutable - in other words:

* it gives you an accumulator (the result so far) as well as the current value,
* and you return the new value of the accumulator.

So rather than mutating the accumulator, try to build the new accumulator out of the existing one + current value. This will most likely involve using concat() instead of push().

(This works very similarly to redux 'reducers' by the way, which take a 'state' and an 'action', then return the new 'state' after the action is applied, without actually editing the old one. If you're wondering why immutability is sought after here, it's mostly because it's easier to debug & reason about).

Reply:
Not going to lie, I'm not sure how to implement these changes and also not exactly sure why, as it's working as intended. But: I'm open to making changes if you could give a little more of an explanation on why its better to use concat rather than push

Reply:
https://github.com/patrickbrett/mlvet/pull/54/commits/95a6f7be59fa28c3ebc165d1ab048625e448506f